src/dhtproto/node/neo/request/GetAll.d

```diff
@@ -231,16 +227,16 @@ public abstract scope class GetAllProtocol_v0
         Gets the next record in the iteration, if one exists.
 
         Params:
-            key = receives the key of the next record, if available
-            value = receives the value of the next record, if available
+            dg = called with the key and value of the next record, if available
 
         Returns:
-            true if a record was returned via the out arguments or false if the
-            iteration is finished
+            true if a record was passed to `dg` or false if the iteration is
+            finished
 
     ***************************************************************************/
 
-    abstract protected bool getNext ( out hash_t key, ref void[] value );
+    abstract protected bool getNext (
+        void delegate ( hash_t key, Const!(void)[] value ) dg );
```

Comment:
I wonder if this delegate should return a boolean indicating if it want's to continue the operation, akin to `opApply`, but I can't think of the compelling reason right now, as unlike `opApply`, user is responsible of calling `getNext` for each next record.

Reply:
(in which case, returning false from the delegate would make getNext return false as well).

Reply:
This delegate's function is just to fetch the current record; it has nothing to do with the decision about whether to continue or not.

Reply:
That's true right now, I'm just wondering if it makes sense to allow it to influence decision if the iteration should continue or not. Probably not, I wouldn't complicate it without any reason :+1:.